### ISSUE TITLE
Add support for passing credentials

### ISSUE THREAD

**Issue by AlessioVallero** [kwojcicki/amazon-dax-client-v3#8]
Hi! Thanks for your work on making this library happening!

My application authenticates using temporary credentials from the credentials provider, which I use to pass to DynamoDB with no issue.

However, it seems that "credentials" or "providerCredentials" is actually not a config item accepted by the TypeScript definition of your library.

Is that something that you already support and just need to expose to the types?

**Comment by kwojcicki** [comment]
Hey @AlessioVallero the DAX library itself reuses the credentials from the passed in DDB client. So I would hope:

```javascript
 const documentDaxClient = new AmazonDaxClient({
        client: DynamoDBDocumentClient.from(new DynamoDBClient({
            endpoint: process.env.dax,
            region: 'us-east-2',
            credentials:  yourTempCreds
        }))
    });
```

Just works.

With that being said I've not experimented with temporary credentials so if **that does not work** can you provide a small repro and I can then add support.

**Comment by AlessioVallero** [comment]
Sure. What I believe is happening, when using temporary credentials, is that `DynamoDBClient` WILL take care of refreshing the credentials. However, the refreshed credentials generated later are not passed along to the AmazonDaxClient like you are doing at the beginning, when the class instance is being created.
So you will need to find a way to pass it back to the DaxClient when credentials are refreshed.

Hard to tell, but maybe the issue would be solved [by decommenting this line](https://github.com/kwojcicki/amazon-dax-client-v3/blob/main/src/AmazonDaxClient.js#L173)?
Though, in my case, I would need it event when `client` is a `DynamoDBDocumentClient`.

Here's a piece of the code I'm running:
```
import { DynamoDBClient, DynamoDBClientConfig } from '@aws-sdk/client-dynamodb'
import { fromTemporaryCredentials } from '@aws-sdk/credential-providers'
import { DynamoDBDocumentClient } from '@aws-sdk/lib-dynamodb'
import AmazonDaxClient from 'amazon-dax-client-sdkv3'

export const createDynamodbClient = (logger: Logger) => {
  const options: DynamoDBClientConfig = {
    region: config.resources.dynamodb.region || undefined,
    endpoint: config.resources.dynamodb.endpoint || undefined,
  }

  const credentials = fromTemporaryCredentials({
    params: {
      RoleArn: 'MY_ROLE_ARN',
      RoleSessionName: 'MY_ROLE_SESSION_NAME',
    },
    clientConfig: {
      region: config.resources.dynamodb.region,
    },
  })
  options.credentials = credentials

  const client = new DynamoDBClient(options)
 
  const marshallOptions = {
    // Whether to automatically convert empty strings, blobs, and sets to `null`.
    convertEmptyValues: false, // false, by default.
    // Whether to remove undefined values while marshalling.
    removeUndefinedValues: false, // false, by default.
    // Whether to convert typeof object to map attribute.
    convertClassInstanceToMap: false, // false, by default.
  }
  const unmarshallOptions = {
    // Whether to return numbers as a string instead of converting them to native JavaScript numbers.
    wrapNumbers: false, // false, by default.
  }
  const translateConfig = { marshallOptions, unmarshallOptions }

  const documentClient = new AmazonDaxClient({
    client: DynamoDBDocumentClient.from(client, translateConfig),
  })

  return {
    client,
    documentClient,
    disconnect: () => {
      client.destroy()
    },
  }
}
```

**Comment by kwojcicki** [comment]
Hey @AlessioVallero apologies for the delay on getting to this.

I've updated the DAX client to refresh the credentials if the credentials are going to expire within the next 5 minutes. This follows the existing pattern within smithy (used by the aws-sdk) https://github.com/smithy-lang/smithy-typescript/blob/main/packages/core/src/util-identity-and-auth/memoizeIdentityProvider.ts#L6-L14

Relevant commits that added this refreshing:
- https://github.com/kwojcicki/amazon-dax-client-v3/commit/c4245104ef6fc428ee5e085dcf09430cbf47c20f
- https://github.com/kwojcicki/amazon-dax-client-v3/commit/7402b7227b7563fdd69e62afb0d41cc79b217399#diff-ece1ddb32eaad252a1f26e20c4098d9352dd37bc97119ecab741e2328950839c


Verified this works via creating a lambda that looks like the following:

tldr of the code:

- initialize temp creds at the top level of the lambda such that its reused across each lambda invocation (https://docs.aws.amazon.com/lambda/latest/dg/nodejs-handler.html)
- create a DAX client using those temp credentials
- perform various DAX operations
- sleep

I ran this lambda for 20 minutes and each call was successful indicating the credentials are refreshed (I also verified this with debug logs that I've since removed)

```javascript

// creds/clients hoisted to the top level such that they are created once during the initial lambda sandbox creation and never again.

const tempCreds = fromTemporaryCredentials({
    params: {
        RoleArn: roleArn,
        DurationSeconds: 900
    },
    clientConfig: {
        useGlobalEndpoint: false,
        endpoint: vpcEndpoint,
        region: 'us-east-2',
    }
});

const documentDaxClient = new AmazonDaxClient({
    client: DynamoDBDocumentClient.from(new DynamoDBClient({
        endpoint: process.env.dax,
        region: 'us-east-2',
        credentials: tempCreds
    })),
    config: {
        connectTimeout: 10_001
    }
});


export const dynamoDBDocumentClientExamples = async () => {
  // omitted for brevity but basically whatever is here https://github.com/kwojcicki/amazon-dax-client-v3/blob/main/test/dynamoDBDocumentClient.ts
}

export const handler = async (_event: any, _context: any) => {
    for (let i = 0; i < 5; i++) {
      await dynamoDBDocumentClientExamples();
      await new Promise(r => setTimeout(r, 60_000))
    }
    return "ok";
};
````
